langchain/document_loaders/blob_loaders/file_system.py

```diff
@@ -1,9 +1,38 @@
 """Use to load blobs from the local file system."""
 from pathlib import Path
-from typing import Iterable, Optional, Sequence, Union
+from typing import Callable, Iterable, Iterator, Optional, Sequence, TypeVar, Union
 
 from langchain.document_loaders.blob_loaders.schema import Blob, BlobLoader
 
+T = TypeVar("T")
+
+
+def _make_iterator(
+    length_func: Callable[[], int], show_progress: bool = False
```

Comment:
out of curiosity can length_func not just be called when passed into this function (so arg would just be `length` instead of length_func)

Reply:
It forces us to iterate through all the files on the file system, which is unnecessary if we're not using tqdm to show the progress. So we pass a function so we can calculate the length on the condition that we want to use tqdm

Reply:
now that's big brain 😻